product_listings_manager/products.py

```diff
@@ -260,14 +260,13 @@ def compose_get_dbh():
         return pgdb.connect(database=dbname, host=dbhost, user=dbuser, password=dbpasswd)
     compose_get_dbh = staticmethod(compose_get_dbh)
 
-    def get_module(compose_dbh, name, stream, version):
-        qargs = dict(name=name, stream=stream, version=version)
+    def get_module(compose_dbh, name, stream):
+        qargs = dict(name=name, stream=stream)
         qry = """
             SELECT id
             FROM modules
             WHERE name = %(name)s
             AND stream = %(stream)s
-            AND version = %(version)s
```

Comment:
A result will have multiple results after removing this condition. Will it affect anything?

Reply:
Yes, name+stream+version is unique. There are already couple of records with same name+stream value.

Also, you still need the version (and even context) to make the API request to plm, right? (Brew requires full module build name.)

Reply:
Is it necessary to order by id or version to newer record?

Reply:
I think this PR is half of the solution.  Lubomir is correct that we want the same product mapping listing regardless of version.  It's only name and stream  that matter.  The problem is that modules table still has a version column and the importtree.py script is still populating that.  So, we need this PR *and* we need to drop the version column and update the import scripts.  I'd like to have someone else confirm that.

@hluk to your point, yes, you still need version and context when making the API request because we need tat information to query the brew build.  But once we have the brew build, the mappings are based on just name and stream.

Reply:
@dgregor I would consider the version still useful for tracking historical composes that may be deleted. Compare to regular RPM mappings: from each compose full NVR is imported, even though only name is used for product listings.

Reply:
@lubomir I'm not opposed to tracking the version information.  But as-is precalc_module_trees is joining on "tree_modules.modules_id = %(module_id)d" and the module_id value is determined from the call to get_module.  If you have a separate entry in the modules table for each version of a module stream, you don't know which module id value you're going to get when you get get_module.  If it's not the id that matches the most recent import then precalc_module_trees could return the wrong set of trees.

Reply:
That said, I think it would only be an issue when a module stream moves out of a product.  It's a use case that we need to consider, but is an edge case that could be addressed separately.

Reply:
That's a good point. Maybe the logic should be adjusted a bit: the call could first find latest tree for the given product, and only then check if the module name:stream combination is in there. That seems to be closer to how RPMs are handled too.